src/http/client.cc

```diff
@@ -341,13 +341,11 @@ future<> client::do_make_request(connection& con, request& req, reply_handler& h
         connection::reply_ptr reply = co_await con.do_make_request(req);
         auto& rep = *reply;
         if (expected.has_value() && rep._status != expected.value()) {
-            if (!http_log.is_enabled(log_level::debug)) {
-                throw httpd::unexpected_status_error(reply->_status);
+            if (http_log.is_enabled(log_level::debug)) {
```

Comment:
i'd prefer early return for less indent. but don't feel strong either way.

Reply:
This is to de-duplicate two throw-s of the same exception